openHarmony/openHarmony_file.js

```diff
@@ -208,94 +208,103 @@ Object.defineProperty($.oFolder.prototype, 'content', {
 
 
 /**
- * Lists the file names contained inside the folder.
- * @param   {string}   [filter]               Filter wildcards for the content of the folder.
- *
- * @returns {string[]}   The names of the files contained in the folder that match the filter.
+ * Enum for the type of content to retrieve from the oFolder.
+ * @enum {QFlag}
  */
-$.oFolder.prototype.listFiles = function(filter){
-    if (typeof filter === 'undefined') var filter = "*";
-
-    var _dir = new QDir;
-    _dir.setPath(this.path);
-    if (!_dir.exists) throw new Error("can't get files from folder "+this.path+" because it doesn't exist");
-    _dir.setNameFilters([filter]);
-    _dir.setFilter( QDir.Files);
-    var _files = _dir.entryList();
-
-    return _files;
+const _oFolderContentType = {
+  FOLDER: QDir.Filters(QDir.Dirs | QDir.NoDotAndDotDot),
+  FILE: QDir.Files
 }
 
 
 /**
- * get the files from the folder
- * @param   {string}   [filter]     Filter wildcards for the content of the folder.
+ * Lists the contents of the folder, filtered by the contentType and name filter(s).
+ * @param   {$.oFolder}            [path]             Path to the folder.
+ * @param   {_oFolderContentType}  [contentType]      Type of content to retrieve.
+ * @param   {string|string[]}      [filter="*"]       Single filter, or array of filters for the contents of the folder.
  *
- * @returns {$.oFile[]}   A list of files contained in the folder as oFile objects.
- */
-$.oFolder.prototype.getFiles = function( filter ){
-    if (typeof filter === 'undefined') var filter = "*";
-    // returns the list of $.oFile in a directory that match a filter
+ * @returns {string[]}   Names of the folder contents that match the filter and type provided.
+ */
+function _oListFolderContents(path, contentType, filter) {
```

Comment:
because of the way openHarmony gets loaded, we can get issues with simple scoped functions that aren't attached to one of the classes. But I think we can turn this private function into a function we actually might want to use as a user, why not call it for example like this?

```javascript
$.oFolder.prototype.getEntries= function(path, contentType, filter)
```

Reply:
Resolved with: d1dc5c70aa5ae3d7d9da9675078473c4b23ddfeb
Moved into oFolder and renamed to listEntries (since get is used to return oFolder/oFile instances elsewhere and this will only ever return string[]).
Added docstrings as discussed that provides information on calling the function directly with examples.